mesonbuild/mesonlib/universal.py

```diff
@@ -2104,13 +2104,18 @@ def __eq__(self, other: object) -> bool:
         return NotImplemented
 
     def __lt__(self, other: object) -> bool:
-        if isinstance(other, OptionKey):
-            return (
-                self.name < other.name and
-                self.subproject < other.subproject and
-                self.machine < other.machine and
-                self.lang < other.lang)
-        return NotImplemented
+        if not isinstance(other, OptionKey):
+            return NotImplemented
+        elif self.subproject != other.subproject:
+            return self.subproject < other.subproject
+        elif self.type != other.type:
+            return self.type < other.type
+        elif self.lang != other.lang:
+            return self.lang < other.lang
+        elif self.machine != other.machine:
+            return self.machine < other.machine
+        else:
+            return self.name < other.name
```

Comment:
I think you can use `(self.subproject, self.type, self.lang, self.machine, self.name) < (other.subproject, other.type, other.lang, other.machine, other.name)` here.

Reply:
oh nice, didn't know that syntax, thanks. Updated.